drivers/rtc/rtc_ll_stm32.c

```diff
@@ -149,16 +176,87 @@ static int rtc_stm32_get_time(const struct device *dev, struct rtc_time *timeptr
 	timeptr->tm_min = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_MINUTE(rtc_time));
 	timeptr->tm_sec = __LL_RTC_CONVERT_BCD2BIN(__LL_RTC_GET_SECOND(rtc_time));
```

Comment:
tm_wday computation is working but not consistent with values given in reference manual, I hesitated to fix it but it is probably to much nitpicking.

Reply:
Could you expand on this? From what I can tell from the [reference manual for stm325858ai](https://www.st.com/resource/en/reference_manual/rm0456-stm32u5-series-armbased-32bit-mcus-stmicroelectronics.pdf) page 2599 the RTC_DR.WDU goes from 1 (monday) to 7 (sunday), while the [struct tm](https://cplusplus.com/reference/ctime/tm/) goes from 0 (monday) to 6 (sunday). so RTC_DR.WDU - 1 seems consistent with at least this reference manual :)

Reply:
never mind, it seems you are right, 0 in `struct tm` is sunday, of course, as long as the offset is the same it will work as you pointed out, I say fix it :)

Reply:
fixed by [804d899](https://github.com/zephyrproject-rtos/zephyr/pull/62304/commits/804d899d594f98af309a8ab143663df586d0782f)